annotation/util.scm

```diff
@@ -388,7 +388,7 @@
       (run-query
         (BindLink
           (VariableNode "$loc")
-          (AndLink
+          (ContextLink
```

Comment:
@linas its because we have such a pattern in the dataset that in a Context of a given Reactome pathway, the genes/molecules location can be different. We used to represent such a pattern by AndLink as:
```

(AndLink
  (MemberLink (GeneNode "KNL1") (ConceptNode "R-HSA-606279"))

  (EvaluationLink
    (PredicateNode "has_location")
    (ListLink (GeneNode "KNL1") (ConceptNode "nucleoplasm"))))

(AndLink
  (MemberLink (GeneNode "KNL1") (ConceptNode "R-HSA-141444"))
  
  (EvaluationLink
  (PredicateNode "has_location")
  (ListLink (GeneNode "KNL1") (ConceptNode "cytosol"))))
```
But that is incorrect, because the AndLink during the PatternMatching is used inorder to look for morethan one patterns (regardless of their context) and we end up having incorrect results. We have changed the AndLink into ContextLink and results are correct now.

Reply:
Ahhh! OK, that actually makes sense!  That's a good solution.

There's actually another solution possible, without any change to `ContextLink`; you could have searched for `(PresentLink (AndLink ... stuff..))` because `PresentLink` tells the pattern matcher to look for the stuff that it wraps.  Even now, saying `(PresentLink (ContextLink ...stuff..))` would help make it clear that you are searching for a single `ContextLink`